### ISSUE TITLE
Leverage cl-https-everywhere for force-https-mode

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nyxt#1303]
Not all URLs should be rewritten to HTTPS, and cl-https-everywhere can help us with that:
https://github.com/ruricolist/cl-https-everywhere

**Comment by hendursaga** [comment]
Isn't the EFF retiring HTTPS Everywhere? We might want to look into DuckDuckGo's Smarter Encryption extension, or [at least its list](https://github.com/duckduckgo/smarter-encryption#just-want-the-list).

**Comment by Ambrevar** [comment]
Right, I found the announcement: https://www.eff.org/deeplinks/2021/04/https-everywhere-now-uses-duckduckgos-smarter-encryption

**Comment by hendursaga** [comment]
Would the list being licensed CC BY-NC-SA 4.0 be a problem? Perhaps by not bundling but allowing the option to download it??

**Comment by hendursaga** [comment]
Obviously we'd want to have an option, perhaps even not the default, since the [compressed file](https://staticcdn.duckduckgo.com/https/smarter_encryption_latest.tgz) is 1/4 gigabyte..

**Comment by hendursaga** [comment]
So the file literally is just a bazillion hostnames, one line each. I was thinking some sort of succinct data structure, like they use for dictionary lookup? Or do we just want to throw it into an SQLite database?

**Comment by Ambrevar** [comment]
If it's this big, SQLite might be a good option.

**Comment by hendursaga** [comment]
What SQL / SQLite library should we use? [cl-sqlite](https://cl-sqlite.common-lisp.dev/), perhaps?

**Comment by hendursaga** [comment]
And perhaps some query generator library like [sxql](https://github.com/fukamachi/sxql)??

**Comment by aadcg** [comment]
I think this is  the topic of an extension.
